### ISSUE TITLE
Python 3.5 support?

### ISSUE THREAD

**Issue by ahartikainen** [pymc-devs/pymc#3465]
If you have questions about a specific use case, or you are not sure whether this is a bug or not, please post it to our discourse channel: https://discourse.pymc.io  

## Description of your problem

Hi, is Python 3.5 supported anymore?

```python
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:664: in _load_unlocked
    ???
<frozen importlib._bootstrap>:634: in _load_backward_compatible
    ???
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:299: in load_module
    six.exec_(co, mod.__dict__)
arviz/tests/test_plots.py:8: in <module>
    import pymc3 as pm
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/pymc3/__init__.py:10: in <module>
    from .model_graph import model_to_graphviz
E     File "/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/pymc3/model_graph.py", line 50
E       vars: MutableSet[RV] = set(self.var_list)
E           ^
E   SyntaxError: invalid syntax
```

**Please provide any additional information below.**
This happened after https://github.com/pymc-devs/pymc3/commit/9f3bba04b15b9e0b046e651b63b82b93146dcae3

## Versions and main components

* PyMC3 Version: 3.6 git Master
* Theano Version: 1.0.4
* Python Version: Python 3.5.6
* Operating system: ArviZ Docker (conda/miniconda3)
* How did you install PyMC3: pip

**Comment by lucianopaz** [comment]
Strange, type hints [should be fully](https://docs.python.org/3/library/typing.html) supported in 3.5.6. Is this blocking arviz's CI?

**Comment by ahartikainen** [comment]
Yes

https://travis-ci.org/arviz-devs/arviz/builds/528762356

It might be that we have some old package somewhere.

**Comment by lucianopaz** [comment]
Actually, the error is happening because of a [variable annotation](https://www.python.org/dev/peps/pep-0526/), not a type hint. That was introduced in python 3.6. I'll make a quick fix so that Arviz CI can still build properly.

**Comment by lucianopaz** [comment]
@ahartikainen, thanks for reporting this! Our CI environment only tests on python 3.6, so we hadn't noticed this during testing.

**Comment by lucianopaz** [comment]
Should be fixed by #3466
